upup/pkg/fi/http.go

```diff
@@ -91,7 +91,7 @@ func downloadURLAlways(url string, destPath string, dirMode os.FileMode) error {
 				KeepAlive: 30 * time.Second,
 			}).DialContext,
 			TLSHandshakeTimeout:   10 * time.Second,
-			ResponseHeaderTimeout: 10 * time.Second,
+			ResponseHeaderTimeout: 30 * time.Second,
```

Comment:
Maybe we should use the default transport directly

https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/net/http/transport.go;bpv=1;bpt=1;l=43?gsn=DefaultTransport&gs=kythe%3A%2F%2Fgolang.org%3Flang%3Dgo%3Fpath%3Dnet%2Fhttp%23var%2520DefaultTransport
httpClient := &http.Client{
		Transport: http.DefaultTransport
},

I can't remember where these values are coming, if there are no special reasons and the default are more conservative we should use them

Reply:
Looks like this was implemented to support slow downloads - https://github.com/kubernetes/kops/commit/19f47d49984777b9886fa159967716909bd6b38b

Although I'm unsure of the history around this

Reply:
I did it, but I can't remember where did  I get those values, most probably from some kubernetes code, but they don't seem to apply here so I think that better to use the more conservative values from the default client, that in this case doesn't have any timeout at all for response header.

```
	httpClient := &http.Client{
		Transport: &http.Transport{
	Proxy: ProxyFromEnvironment,
	DialContext: (&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
	}).DialContext,
	ForceAttemptHTTP2:     true,
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
		},
	}

Reply:
I think the goal was to reduce the IdleConnTimeout mostly. Not sure if `ForceAttemptHTTP2` affected things in any way in the past, but  don't see a real need for in in the case of downloading various binaries.
This change looks pretty good to me.

Reply:
you know better than me the network requirements here, the change lgtm too, just wanted to add more context in case you think some other change will needed

Reply:
I appreciate. Thanks @aojea! 😄